lib/SILOptimizer/Mandatory/TFDeabstraction.cpp

```diff
@@ -1808,12 +1811,16 @@ transformTensorFromScalar(ApplyInst *apply,
     // We use int32 as the element type of the zero-d shape array.
     auto int32Ty =
         ctx.getInt32Decl()->getDeclaredType()->getCanonicalType();
-    assert(it->second.getKind() == SymbolicValue::Address);
 
-    SmallVector<unsigned, 4> accessPath;
-    auto *scalarMemoryObject = it->second.getAddressValue(accessPath);
-    assert(accessPath.empty());
-    auto scalarValue = scalarMemoryObject->getValue();
+    SymbolicValue scalarValue;
```

Comment:
What bug is it fixing?

Reply:
It used to be that the argument to __tf_tensor_from_scalar was always an address because it was generic. This function now also handles __tf_string_tensor_from_scalar, whose argument is a value because it's concrete. I'll add some comments.